### PR TITLE
Fixing Calendar issues related to DST and event organizer timezone being different than user timezone.

### PR DESCRIPTION
**Pull request by chamakura**
- Does the pull request solve a **related** issue?
Yes

- If so, can you reference the issue?
#2085 

- What does the pull request accomplish? Use a list if needed.
* Change to use node-ical instead of ical for parsing ics files
* Fixing issues because of DST offsets

- If it includes major visual changes please add screenshots.

### CODE REVIEW

File: modules/default/calendar/calendarfetcher.js [hunk 1]
```diff
@@ -70,6 +71,35 @@ const CalendarFetcher = function (url, reloadInterval, excludedEvents, maximumNu
...
+			// based on whether the transition is to DST, or from DST.
+			const getDSTAdjustmentMinutes = function (firstTimeUTC, secondTimeUTC, timeZone) {
+				const isFirstTimeInDST = momentTz.tz(firstTimeUTC.toISOString(), timeZone).isDST();
+				const isSecondTimeInDST = momentTz.tz(secondTimeUTC.toISOString(), timeZone).isDST();
+
+				// Apparently 'Australia/Lord_Howe' is the only timezone where DST is adjusted by 30 minutes
+				// https://www.timeanddate.com/time/dst/#:~:text=You%20set%20your%20clock%20forward,Daylight%20Savings%20or%20Daylight%20Saving%3F
+
+				const dstOffset = timeZone === "Australia/Lord_Howe" ? 30 : 60;
+				if (isFirstTimeInDST == isSecondTimeInDST) {
```
Comment (rejas) [inline modules/default/calendar/calendarfetcher.js:92]: this should be a ===

File: modules/default/calendar/calendarfetcher.js [hunk 2]
```diff
@@ -183,10 +214,25 @@ const CalendarFetcher = function (url, reloadInterval, excludedEvents, maximumNu
...
+						const datesWithoutDSTAdjustment = rule.between(pastLocal, futureLocal, true, limitFunction);
+
+						// The dates returned by 'rule' are offset based on UTC, so in addition to converting them
+						// to local date-time, we also need to adjust for the required DST offsets
+						const dates = datesWithoutDSTAdjustment.map(function (dateWithoutDSTAdjustment) {
+							var eventDate = moment(dateWithoutDSTAdjustment);
+							eventDate.add(getDSTAdjustmentMinutes(startDate, eventDate, localTz), "minutes");
+
+							var localTz = momentTz.tz.guess();
+							if (organizerTz != localTz) {
```
Comment (rejas) [inline modules/default/calendar/calendarfetcher.js:228]: this should be a !==

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,18 +11,23 @@ _This release is scheduled to be released on 2020-10-01._
 
 ### Added
 
+- Added a dependency on 'node-ical' and 'moment-timezone' libraries
```
Comment (rejas) [inline CHANGELOG.md:14]: are there any direct benefits froms witching to node-ical?

Comment (chamakura) [reply CHANGELOG.md:14]: From what I can see, 'ical' is mainly targeted for pure Javascript environments whereas 'node-ical' will provide more flexibility/functionality in Node environments. This was evident when I tried fixing an incorrect issue parsing start-time timezone using 'ical', but couldn't do it without having to copy all of the timezone database data.

File: modules/default/calendar/calendarfetcher.js (hunk 1 above)
Comment (chamakura) [reply modules/default/calendar/calendarfetcher.js]: Done.

File: modules/default/calendar/calendarfetcher.js (hunk 2 above)
Comment (chamakura) [reply modules/default/calendar/calendarfetcher.js]: Done.

**Comment by chamakura** [comment]
PTAL @rejas

**Comment by MichMich** [comment]
@rejas, did you (by any chance) test this?

**Comment by rejas** [comment]
> @rejas, did you (by any chance) test this?

no, just took a quick look at the code.

**Comment by chamakura** [comment]
@MichMich - any update on whether this change can be merged or not?

**Comment by MichMich** [comment]
Sorry for the slow response. This PR has some conflicts I can't fix. Would it be possible four you to fix these conflicts?

**Comment by chamakura** [comment]
I will take a look this weekend.

**Comment by sdetweil** [comment]
with my all day changes (already merged) and the updated node-ical node library, we don't need this fix. the problem was in ical library..

I am waiting for the node-ical author to publish to node library before submitting pr  

a sample using ical.js library is listed at the end of #1798 
its expected that ical.js will reject this change (also submitted) due to requirement to be in browser only

**Comment by MichMich** [comment]
Ok, so just to be sure: I can close this PR?

**Comment by sdetweil** [comment]
u should let him confirm, but I believe yes

**Comment by chamakura** [comment]
It's ok to close this PR if the fixes are already in. I had a separate DST-related fix as well, but will see if it's address based on the other ical fixes - if not I'll submit a new PR.

**Comment by sdetweil** [comment]
I also want to move to node-ical, as ical.js is browser only. I put all the timezone fixes in the node-ical module, where they should be.
I also added support for ms timezone names and offset timezones
moment-tz does all the DST adjustment, so mm doesn't have to

**Comment by MichMich** [comment]
Ok. Closing for now. Thanks for all your hard work, people!

**Pull request closed without merging**